File: Mobile School/app/src/main/java/bf/ujkz/schoolconnect/models/Payment.kt
package bf.ujkz.schoolconnect.models

data class Payment(
    val id: Int,
    val amount: String,
    val payment_date: String,
    val payment_method: String,
    val receipt_number: String,
    val remarks: String?,
    val status: String? = null // "validee", "en_attente", "rejetee"
)

data class PaymentRequest(
    val amount: Double,
    val payment_method: String,
    val remarks: String?
)

data class PaymentSummary(
    val student: PaymentStudentMini,
    val school_fees: String,
    val total_paid: Double,
    val remaining_balance: Double,
    val payment_percentage: Double
)

data class PaymentStudentMini(
    val id: Int,
    val full_name: String,
    val `class`: String
)

data class PaymentDetails(
    val id: Int,
    val amount: String,
    val payment_date: String,
    val payment_method: String,
    val receipt_number: String,
    val remarks: String?,
    val status: String? = null,
    val student: PaymentStudentName
)

data class PaymentStudentName(
    val full_name: String
)

data class ReceiptData(
    val receipt_url: String,
    val receipt_number: String
)

File: gestion-scolaire/app/Models/Payment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Payment extends Model
{
    use HasFactory;

    // Statuts possibles d'un paiement
    public const STATUS_EN_ATTENTE = 'en_attente';
    public const STATUS_VALIDEE = 'validee';
    public const STATUS_REJETEE = 'rejetee';

    protected $fillable = [
        'student_id',
        'amount',
        'payment_date',
        'payment_method',
        'receipt_number',
        'remarks',
        'status',
    ];

    protected $casts = [
        'amount' => 'decimal:2',
        'payment_date' => 'date',
    ];

    public function student(): BelongsTo
    {
        return $this->belongsTo(Student::class);
    }

    public function isValidated(): bool
    {
        return $this->status === self::STATUS_VALIDEE;
    }
}